### PR TITLE
Add json-c/0.13.1

### PR DESCRIPTION
**Pull request by theirix**
Specify library name and version:  **json-c/0.13.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/json-c/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.name + "-" + self.version + '-20180305'
+        os.rename(extracted_dir, self._source_subfolder)
+        tools.patch(base_path=self._source_subfolder, patch_file="json-c.patch")
```
Comment (uilianries) [inline recipes/json-c/all/conanfile.py]:
```suggestion
```

File: recipes/json-c/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        if tools.cross_building(self.settings) and self.settings.os != "Windows":
+            host = tools.get_gnu_triplet(str(self.settings.os), str(self.settings.arch))
+            tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                                  "execute_process(COMMAND ./configure ",
+                                  "execute_process(COMMAND ./configure --host %s " % host)
+        cmake = CMake(self)
+        cmake.configure(build_folder=self._build_subfolder)
+        return cmake
+
+    def build(self):
```
Comment (uilianries) [inline recipes/json-c/all/conanfile.py]:
```suggestion
    def build(self):
        tools.patch(**self.conan_data["patches"][self.version])
```

File: recipes/json-c/all/conandata.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "0.13.1":
+    url: "https://github.com/json-c/json-c/archive/json-c-0.13.1-20180305.tar.gz"
+    sha256: "5d867baeb7f540abe8f3265ac18ed7a24f91fe3c5f4fd99ac3caba0708511b90"
```
Comment (uilianries) [inline recipes/json-c/all/conandata.yml]:
```suggestion
    sha256: "5d867baeb7f540abe8f3265ac18ed7a24f91fe3c5f4fd99ac3caba0708511b90"
patches:
  "0.13.1":
     - patch_file: "json-c.patch"
       base_path: "source_subfolder"
```

Comment (theirix) [reply recipes/json-c/all/conandata.yml]: Great! Can we add it to the CCI wiki?

**Comment by theirix** [comment]
Syntax with a list that is proposed in review and FAQ is not supported with `tools.patch`. Should we use non-list syntax here?

File: recipes/json-c/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    homepage = "https://github.com/json-c/json-c"
+    license = "MIT"
+    exports_sources = ["CMakeLists.txt", "json-c.patch"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "use_urandom": [True, False],
+        "use_windows_cryptoapi": [True, False]
```
Comment (danimtb) [inline recipes/json-c/all/conanfile.py]: Where is this option used?

File: recipes/json-c/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/json-c/json-c"
+    license = "MIT"
+    exports_sources = ["CMakeLists.txt", "json-c.patch"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "use_urandom": [True, False],
```
Comment (danimtb) [inline recipes/json-c/all/conanfile.py]: unused?

File: recipes/json-c/all/conanfile.py (hunk 4 above)
Comment (theirix) [reply recipes/json-c/all/conanfile.py]: Yes, copy-pasted from jansson. Removed.

File: recipes/json-c/all/conanfile.py (hunk 5 above)
Comment (theirix) [reply recipes/json-c/all/conanfile.py]: Removed

File: recipes/json-c/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        'shared': False,
+        'fPIC': True
+    }
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.name + "-" + self.version + '-20180305'
```
Comment (uilianries) [inline recipes/json-c/all/conanfile.py]:
```suggestion
        url = self.conan_data["sources"][self.version]["url"]
        extracted_dir = url[url.rfind('/')+1:url.find(".tar.gz")]
```
`20180305` should not be hardcoded.

Comment (theirix) [reply recipes/json-c/all/conanfile.py]: Good idea, thanks!

**Pull request merged**